src/vm/types/mod.rs

```diff
@@ -342,7 +342,7 @@ impl SequenceData {
         macro_rules! drain_filter {
             ($data:expr, $seq_type:ident) => {
                 let mut i = 0;
-                while i != $data.data.len() {
+                while i < $data.data.len() {
```

Comment:
This change shouldn't be necessary, right? The while loop can only ever either remove an element (decreasing len() by 1) or increment i, meaning that `i` should never cross `.len()`. I'm just a little wary of changing something like this (a loop condition deep in the type code) unless it's absolutely necessary.

Reply:
Sure, I'll remove this.